src/Build/FileSystem/MSBuildFileSystemBase.cs

```diff
@@ -14,66 +15,95 @@ namespace Microsoft.Build.FileSystem
     /// - must be thread safe
     /// - may cache some or all the calls.
     /// </summary>
-    public abstract class MSBuildFileSystemBase
+    public class MSBuildFileSystemBase : IFileSystem
     {
+        private IFileSystem _defaultFileSystem;
+        private IFileSystem DefaultFileSystem
+        {
+            get
+            {
+                if (_defaultFileSystem == null)
+                {
+                    var newDefaultFileSystem = new CachingFileSystemWrapper(FileSystems.Default);
+                    System.Threading.Interlocked.CompareExchange(ref _defaultFileSystem, newDefaultFileSystem, null);
+                }
+                return _defaultFileSystem;
+            }
+        }
+
+        public MSBuildFileSystemBase()
+        { }
+
+        internal MSBuildFileSystemBase(IFileSystem defaultFileSystem)
```

Comment:
So is there no way to set the file system from outside MSBuild? Wondering if that might be a useful feature if people have their own file watchers, etc.

Reply:
IFileSystem is private and should be kept that way, no sense making it public.